### PR TITLE
Merge v2 upstream

### PR DESCRIPTION
**Pull request by rbarlow**
This pull request merges in support for Docker v2 to the master branch. All tests pass, though the coverage has dropped from 100% to 96%. Due to this, I have also submitted this pull request:

https://github.com/pulp/pulp_packaging/pull/77

All of the commits in this PR have already been reviewed, except for the merge commit itself. The merge was difficult, and thus the merge commit does significantly alter the new (and old) code.

### CODE REVIEW

File: extensions_admin/test/unit/extensions/admin/test_pulp_cli.py [hunk 1]
```diff
@@ -1,15 +1,19 @@
 import unittest
 
 import mock
+
```
Comment (asmacdo) [inline extensions_admin/test/unit/extensions/admin/test_pulp_cli.py]: remove

File: extensions_admin/test/unit/extensions/admin/test_pulp_cli.py [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
 from pulp.client.commands.repo.cudl import CreateRepositoryCommand, DeleteRepositoryCommand
 from pulp.client.commands.repo.cudl import UpdateRepositoryCommand
 from pulp.client.commands.repo.sync_publish import PublishStatusCommand,\
     RunPublishRepositoryCommand, RunSyncRepositoryCommand
 from pulp.client.commands.repo.upload import UploadCommand
 from pulp.client.extensions.core import PulpCli
+from pulp.client.extensions.extensions import PulpCliSection
+
```
Comment (asmacdo) [inline extensions_admin/test/unit/extensions/admin/test_pulp_cli.py]: this newline too

File: extensions_admin/test/unit/extensions/admin/test_pulp_cli.py [hunk 3]
```diff
@@ -25,7 +29,7 @@ def test_structure(self):
         # create the tree of commands and sections
         pulp_cli.initialize(context)
 
-        # verify that sections exist and have the right commands
+        # verify that sections exist and have the right commands and sections
```
Comment (asmacdo) [inline extensions_admin/test/unit/extensions/admin/test_pulp_cli.py]: s/and sections//

**Comment by rbarlow** [comment]
@asmacdo, I submitted https://github.com/pulp/pulp_docker/pull/105 to respond to the comments on this PR.

**Pull request merged**